paddlenlp/experimental/transformers/gpt/modeling.py

```diff
@@ -0,0 +1,464 @@
+                                self.hidden_size,
+                            ]
+                        )
+                        .astype(dtype)
+                    )
+                elif k.endswith("self_attn.qkv_proj.bias"):
+                    self.transformer_block.qkv_biases[idx].set_value(
+                        v.reshape(
+                            [
+                                self.num_attention_heads // self.config.tensor_parallel_degree,
+                                3,
+                                self.hidden_size // self.num_attention_heads,
+                            ]
+                        )
+                        .transpose([1, 0, 2])
+                        .reshape(
+                            [
+                                self.num_attention_heads
+                                // self.config.tensor_parallel_degree
+                                * 3
+                                * self.hidden_size
+                                // self.num_attention_heads
+                            ]
+                        )
+                        .astype(dtype)
+                    )
+                elif k.endswith("self_attn.out_proj.weight"):
+                    self.transformer_block.linear_weights[idx].set_value(v.astype(dtype))
+                elif k.endswith("self_attn.out_proj.bias"):
+                    self.transformer_block.linear_biases[idx].set_value(v.astype(dtype))
+                elif k.endswith("norm2.weight"):
+                    self.transformer_block.ffn_ln_scales[idx].set_value(v.astype("float32"))
+                elif k.endswith("norm2.bias"):
+                    self.transformer_block.ffn_ln_biases[idx].set_value(v.astype("float32"))
+                elif k.endswith("linear1.weight"):
+                    self.transformer_block.ffn1_weights[idx].set_value(v.astype(dtype))
+                elif k.endswith("linear1.bias"):
+                    self.transformer_block.ffn1_biases[idx].set_value(v.astype(dtype))
+                elif k.endswith("linear2.weight"):
+                    self.transformer_block.ffn2_weights[idx].set_value(v.astype(dtype))
+                elif k.endswith("linear2.bias"):
+                    self.transformer_block.ffn2_biases[idx].set_value(v.astype(dtype))
+                else:
+                    raise ValueError("Unknow weight {}".format(k))
+
+
+class GPTForCausalLMInferenceModel(GenerationInferenceModel, GPTPretrainedModel):
+    """
+    Dynamic Batching for GPT Model with pretraining tasks on top.
+    """
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.gpt = GPTInferenceModel(config)
+
+    @classmethod
+    def from_pretrained(
+        cls, pretrained_model_name_or_path, from_hf_hub: bool = False, subfolder: str | None = None, *args, **kwargs
+    ):
+        # TODO: Support safetensors loading.
+        kwargs["use_safetensors"] = False
```

Comment:
```suggestion
        kwargs["use_safetensors"] = kwargs.get("use_safetensors", False)
```

建议使用这个，因为单分片 safetensors 是可以支持 inferencemodel 加载的。